### PR TITLE
[Quickbooks] Create sales receipt action

### LINKED ISSUE #6031: [ACTION] Create Sales Receipt for Quickbooks Online
**Is there a specific app this action is for?**
Intuit Quickbooks Online
**Please provide a link to the relevant API docs for the specific service / operation.**
https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/salesreceipt#create-a-salesreceipt

### PR DESCRIPTION
**Pull request by andrewjschuang**
Resolves #6031.

### <samp>🤖 Generated by Copilot at 152195b</samp>

This pull request adds a new action component for creating sales receipts in Quickbooks using the `@pipedream/quickbooks` package. It also updates the package version in `components/quickbooks/package.json`.

### <samp>🤖 Generated by Copilot at 152195b</samp>

> _`createSalesReceipt`_
> _New action for Quickbooks_
> _Autumn leaves fall fast_

### <samp>🤖 Generated by Copilot at 152195b</samp>

*  Add a new action component for creating sales receipts in Quickbooks ([link](https://github.com/PipedreamHQ/pipedream/pull/6235/files?diff=unified&w=0#diff-8559f357b02064c2f33699808848be08bc017602c615db52c3514811b267eae1R1-R73))
* Update the version of the `@pipedream/quickbooks` package ([link](https://github.com/PipedreamHQ/pipedream/pull/6235/files?diff=unified&w=0#diff-25c667d24bdd09e573fbea6632373b801df5fca34f2536587619294cef69d430L3-R3))

### CODE REVIEW

File: components/quickbooks/actions/create-sales-receipt/create-sales-receipt.mjs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import quickbooks from "../../quickbooks.app.mjs";
+import { ConfigurationError } from "@pipedream/platform";
+
+export default {
+  key: "quickbooks-create-sales-receipt",
+  name: "Create Sales Receipt",
+  description: "Creates a sales receipt. [See docs here](https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/salesreceipt#create-a-salesreceipt)",
```
Comment (dylburger) [inline components/quickbooks/actions/create-sales-receipt/create-sales-receipt.mjs:7]:
```suggestion
  description: "Creates a sales receipt. [See the documentation](https://developer.intuit.com/app/developer/qbo/docs/api/accounting/all-entities/salesreceipt#create-a-salesreceipt)",
```

Comment (dylburger) [reply components/quickbooks/actions/create-sales-receipt/create-sales-receipt.mjs:7]: Changed 'docs' to 'documentation' to follow Google's style guide.

File: components/quickbooks/actions/create-sales-receipt/create-sales-receipt.mjs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    try {
+      if (typeof (this.lineItems) === "string") {
+        this.lineItems = JSON.parse(this.lineItems);
+      } else {
+        this.lineItems = this.lineItems.map((lineItem) => typeof lineItem === "string"
+          ? JSON.parse(lineItem)
+          : lineItem);
+      }
+    } catch (error) {
+      throw new ConfigurationError(`We got an error trying to parse the LineItems. Error: ${error}`);
```
Comment (dylburger) [inline components/quickbooks/actions/create-sales-receipt/create-sales-receipt.mjs:53]:
```suggestion
      throw new ConfigurationError(`An error occurred while trying to parse the LineItems. Error: ${error}`);
```

Comment (dylburger) [reply components/quickbooks/actions/create-sales-receipt/create-sales-receipt.mjs:53]: Reworded the error message to follow Google's style guide.

**Review by alysonturing — DISMISSED** [review]
LGTM! I'm approving but I believe that the attached suggestions worth to be merged.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Quickbooks_Create_sales_receipt_action_6235_1149.pdf](https://github.com/PipedreamHQ/pipedream/files/11417790/Quickbooks_Create_sales_receipt_action_6235_1149.pdf)

**Review by alysonturing — APPROVED** [review]

**Pull request merged**